### PR TITLE
Add request-specific validation of jsonrpc request params

### PR DESCRIPTION
**Pull request by bitwiseguy**
We already had some validation to ensure the jsonrpc request is properly formatted according to jsonrpc spec 2.0. This PR adds validation for the "params" field within the jsonrpc request, which is different for each jsonrpc method.

### CODE REVIEW

File: client_test/crash_test.go [hunk 1]
```diff
@@ -84,7 +84,7 @@ func TestCrashTolerance(t *testing.T) {
 }
 
 func directlyDefundALedgerChannel(t *testing.T, alpha client.Client, beta client.Client, channelId types.Destination) {
-	id := alpha.CloseLedgerChannel(channelId)
+	id, _ := alpha.CloseLedgerChannel(channelId)
```
Comment (kerzhner) [inline client_test/crash_test.go:87]: We could check the returned error and fail the test on error. This comment also applies to other changes in test files.

**Review by kerzhner — APPROVED** [review]
Looks great! Are you planning on adding unit or integration tests for the new errors?

File: client_test/crash_test.go (hunk 1 above)
Comment (bitwiseguy) [reply client_test/crash_test.go:87]: Good call. Fixed each of these tests to inspect the returned error

**Comment by bitwiseguy** [comment]
> Looks great! Are you planning on adding unit or integration tests for the new errors?

Pushed a commit with some rpc request validation unit tests

**Pull request merged**